app/main/views/suppliers.py

```diff
@@ -263,6 +264,45 @@ def edit_supplier_organisation_size():
     return render_template('suppliers/edit_supplier_organisation_size.html', form=form)
 
 
+@main.route('/trading-status/edit', methods=['GET', 'POST'])
+@login_required
+def edit_supplier_trading_status():
+    form = CompanyTradingStatusForm()
+
+    if request.method == 'POST':
+        if form.validate_on_submit():
+            data_api_client.update_supplier(supplier_id=current_user.supplier_id,
+                                            supplier={"tradingStatus": form.trading_status.data},
+                                            user=current_user.email_address)
+            return redirect(url_for('.supplier_details'))
+
+        current_app.logger.warning(
+            "supplieredit.fail: trading-status:{tstatus}, errors:{tstatus_errors}",
+            extra={
+                'tstatus': form.trading_status.data,
+                'tstatus_errors': ",".join(form.trading_status.errors)
+            })
+
+        return render_template("suppliers/edit_supplier_trading_status.html", form=form), 400
+
+    supplier = data_api_client.get_supplier(current_user.supplier_id)['suppliers']
+
+    prefill_trading_status = None
+
+    if supplier.get('tradingStatus'):
+        if supplier['tradingStatus'] in map(lambda x: x['value'], form.OPTIONS):
+            prefill_trading_status = supplier['tradingStatus']
+
+        else:
+            # The last option in the form is 'other', which we'll use as a fallback if a value exists but doesn't match
```

Comment:
But we're going to have fixed these with a data migration?  Still nice to have the fallback I guess, but I wonder if it's better to not prefill at all in the case we don't have a match, to force people to make a fresh choice?

Reply:
https://github.com/alphagov/digitalmarketplace-scripts/pull/198

Reply:
Yeah, so now the script has run we have no data in the database that doesn't match one of these options.  So I don't think we need to force the prefill of "other" at all - better to make people choose an option if we don't have anything for them.

Reply:
Yep nice, removed.